src/profile/v100/app_preferences/asset_preferences/asset_preferences.rs

```diff
@@ -0,0 +1,162 @@
+use crate::prelude::*;
+use core::hash::Hash;
+use std::hash::Hasher;
+
+/// User off-ledger preferences regarding assets.
+#[derive(
+    Debug,
+    Default,
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    Clone,
+    derive_more::Display,
+    uniffi::Record,
+)]
+#[serde(rename_all = "camelCase")]
+#[display("assets: {:#?}", self.assets)]
+pub struct AssetPreferences {
+    #[serde(default)]
+    pub assets: HashMap<AssetAddress, AssetPreference>,
```

Comment:
should be `IndexMap` else we loose insertion order.

Otherwise might be worth doing like we do with most collection: let `AssetAddress` impl `Identifiable` and use the macro to declare a collection of `AssetPreferences` plural. And use that here.

Reply:
oh right this whole thing should probably be `decl_identified_vec_of(AssetPreference)` which creates a `AssetPreferences` which is serde transparent, so it will be `[]` in JSON. Which is pretty nice.